### PR TITLE
Build docker images in Circle CI

### PR DESCRIPTION
**Pull request by sargun**
* Use the build docker images from Circle CI builds
* Add code coverage metrics
* Try to collect all of the image builds in one place

### CODE REVIEW

**Comment by sargun** [comment]
This again requires the "inside" and "outside" GO PATHs to match, but things break under go 1.10 caching (if we share it), and we have a different "inside" and "outside" set of paths.

File: hack/test-images/ubuntu-systemd-bionic/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM ubuntu:bionic
```
Comment (sargun) [inline hack/test-images/ubuntu-systemd-bionic/Dockerfile]: This was being built by hand, and uploaded by hand before.

File: hack/test-images/ubuntu-systemd-xenial/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+FROM ubuntu:xenial
```
Comment (sargun) [inline hack/test-images/ubuntu-systemd-xenial/Dockerfile]: This was being built by hand, and uploaded by hand before.

File: hack/test-images/build.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
```
Comment (sargun) [inline hack/test-images/build.sh]: This is a combined way to build all the build images.

File: hack/ci-builder/Dockerfile [hunk 4]
```diff
@@ -44,4 +44,5 @@ ENV GOPATH /go
 ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
```
Comment (sargun) [inline hack/ci-builder/Dockerfile]: goveralls is to push the reports

File: executor/mock/standalone/standalone_test.go [hunk 5]
```diff
@@ -43,14 +35,16 @@ type testImage struct {
 }
 
 var (
+	// TODO: Determine how this got built, and add it to the auto image builders?
```
Comment (sargun) [inline executor/mock/standalone/standalone_test.go]: These are all bumped to automated build versions.

File: executor/mock/standalone/standalone.go [hunk 6]
```diff
@@ -1 +1,26 @@
 package standalone
+
```
Comment (sargun) [inline executor/mock/standalone/standalone.go]: :(

File: .buildkite/pipeline.yml [hunk 7]
```diff
@@ -1,27 +1,6 @@
...
-              pushd $$GOPATH/src/github.com/Netflix/titus-executor
-
-              go get -u github.com/alecthomas/gometalinter
-              gometalinter --install
-              go get -u github.com/kardianos/govendor
-
-              make clean
-              make --output-sync -j16 builder all 2>&1 | tee build.log
-    artifact_paths: "*.log;build/distributions/*.deb;build/distributions/*.tar;build/distributions/*.tar.gz"
+  - command: ./.buildkite/build.sh
```
Comment (sargun) [inline .buildkite/pipeline.yml]: This got moved into a little script

**Comment by fabiokung** [comment]
LGTM

@sargun I'm still unsure and :-1: on bind mounting `GOPATHs`, and why we need to share the same cache inside and outside the build containers. I'd rather they have separate caches, they may even be different OSes and arch inside and outside. The build containers can use volumes from data-only container to cache things across builds if it needs to be sped up.

**Comment by sargun** [comment]
Getting rid of the data only containers, and sharing caches with the host builds sped things up significantly. Initially doing the test-local, which built the build cache, and then mounting the GOPATH made build times cut down by almost 50%.

**Comment by sargun** [comment]
If the GOPATH wasn't the same inside and outside of the container, then it wouldn't use the GOPATH. Go seems to freak out whenever it sees bind mounts, or symlinks in gopath. Not sure why.

**Pull request merged**